### PR TITLE
Push7 v2

### PR DESCRIPTION
**Pull request by KwintenSchram**

### CODE REVIEW

File: stack/apps/push7_button/filesystem/button_file.c [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+typedef struct {
+    union {
+        uint8_t bytes[RAW_BUTTON_CONFIG_FILE_SIZE];
+        struct {
+            bool transmit_mask_0;
+            bool transmit_mask_1;
+            bool button_control_menu;
+            bool enabled;
+        } __attribute__((__packed__));
```
Comment (L-I-Am) [inline stack/apps/push7_button/filesystem/button_file.c:44]: at the moment this is 4 bytes but this could be optimised to 4 bits. do we want to do this or is the added complexity not worth it?

File: stack/apps/push7_button/filesystem/button_file.c [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        = { .file_permissions = (file_permission_t) { .guest_read = true, .user_read = true },
+              .file_properties.storage_class = FS_STORAGE_VOLATILE,
+              .length = BUTTON_FILE_SIZE,
+              .allocated_length = BUTTON_FILE_SIZE };
+
+    d7ap_fs_file_header_t permanent_file_header = { .file_permissions
+        = (file_permission_t) { .guest_read = true, .guest_write = true, .user_read = true, .user_write = true },
+        .file_properties.storage_class = FS_STORAGE_PERMANENT,
+        .length = BUTTON_CONFIG_FILE_SIZE,
+        .allocated_length = BUTTON_CONFIG_FILE_SIZE };
```
Comment (L-I-Am) [inline stack/apps/push7_button/filesystem/button_file.c]: allocate more length for permanent files to ensure we can change things in the future without breaking compatibility

File: stack/apps/push7_button/filesystem/button_file.c [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+    ret = d7ap_fs_init_file(BUTTON_FILE_ID, &volatile_file_header, button_file.bytes);
+    if (ret != SUCCESS) {
+        log_print_error_string("Error initializing button file: %d", ret);
+    }
+
+    // d7ap_fs_register_file_modified_callback(BUTTON_CONFIG_FILE_ID, &file_modified_callback);
+    d7ap_fs_register_file_modified_callback(BUTTON_FILE_ID, &file_modified_callback);
+
+    d7ap_fs_read_file(BUTTON_CONFIG_FILE_ID, 0, button_config_file_cached.bytes, &length, ROOT_AUTH);
```
Comment (L-I-Am) [inline stack/apps/push7_button/filesystem/button_file.c]: why do you first read the button config file to test if it's initialised and then read it again? why not read it once?

File: stack/apps/push7_button/filesystem/button_file.c [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+        if ((button_file.mask == true && button_config_file_cached.transmit_mask_1)
+            || (button_file.mask == false && button_config_file_cached.transmit_mask_0))
+            if (button_file_transmit_state && button_config_file_cached.enabled)
+                queue_add_file(button_file.bytes, BUTTON_FILE_SIZE, BUTTON_FILE_ID);
+    }
+}
+
+void button_file_register_cb(ubutton_callback_t callback) { low_level_event_cb = callback; }
+
+void button_file_set_measure_state(bool enable)
```
Comment (L-I-Am) [inline stack/apps/push7_button/filesystem/button_file.c:134]: what's the difference between this and button_file_set_enabled? naming is not that clear

File: stack/apps/push7_button/filesystem/hall_effect_file.c [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+#define DPRINT(...)
+#endif
+
+#define HALL_EFFECT_FILE_ID 52
+#define HALL_EFFECT_FILE_SIZE sizeof(hall_effect_file_t)
+#define RAW_HALL_EFFECT_FILE_SIZE 1
+
+#define HALL_EFFECT_CONFIG_FILE_ID 62
+#define HALL_EFFECT_CONFIG_FILE_SIZE sizeof(hall_effect_config_file_t)
+#define RAW_HALL_EFFECT_CONFIG_FILE_SIZE 2
```
Comment (L-I-Am) [inline stack/apps/push7_button/filesystem/hall_effect_file.c]: this should be 3

File: stack/apps/push7_button/filesystem/hall_effect_file.c [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+        = { .file_permissions = (file_permission_t) { .guest_read = true, .user_read = true },
+              .file_properties.storage_class = FS_STORAGE_VOLATILE,
+              .length = HALL_EFFECT_FILE_SIZE,
+              .allocated_length = HALL_EFFECT_FILE_SIZE };
+
+    d7ap_fs_file_header_t permanent_file_header = { .file_permissions
+        = (file_permission_t) { .guest_read = true, .guest_write = true, .user_read = true, .user_write = true },
+        .file_properties.storage_class = FS_STORAGE_PERMANENT,
+        .length = HALL_EFFECT_CONFIG_FILE_SIZE,
+        .allocated_length = HALL_EFFECT_CONFIG_FILE_SIZE };
```
Comment (L-I-Am) [inline stack/apps/push7_button/filesystem/hall_effect_file.c]: best to allocate more

File: stack/apps/push7_button/filesystem/hall_effect_file.c [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+    hall_effect_file_t hall_effect_file = { .mask = hw_gpio_get_out(HAL_EFFECT_PIN) };
+    d7ap_fs_write_file(HALL_EFFECT_FILE_ID, 0, hall_effect_file.bytes, HALL_EFFECT_FILE_SIZE, ROOT_AUTH);
+}
+
+static void file_modified_callback(uint8_t file_id)
+{
+    if (file_id == HALL_EFFECT_CONFIG_FILE_ID && hall_effect_config_file_transmit_state) {
+        uint32_t size = HALL_EFFECT_CONFIG_FILE_SIZE;
+        d7ap_fs_read_file(HALL_EFFECT_CONFIG_FILE_ID, 0, hall_effect_config_file_cached.bytes, &size, ROOT_AUTH);
+        if (hall_effect_config_file_transmit_state)
```
Comment (L-I-Am) [inline stack/apps/push7_button/filesystem/hall_effect_file.c]: this already gets checked earlier on line 121, not sure if the one on line 121 is necessary

File: stack/apps/push7_button/filesystem/humidity_file.c [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+        = { .file_permissions = (file_permission_t) { .guest_read = true, .user_read = true },
+              .file_properties.storage_class = FS_STORAGE_VOLATILE,
+              .length = HUMIDITY_FILE_SIZE,
+              .allocated_length = HUMIDITY_FILE_SIZE };
+
+    d7ap_fs_file_header_t permanent_file_header = { .file_permissions
+        = (file_permission_t) { .guest_read = true, .guest_write = true, .user_read = true, .user_write = true },
+        .file_properties.storage_class = FS_STORAGE_PERMANENT,
+        .length = HUMIDITY_CONFIG_FILE_SIZE,
+        .allocated_length = HUMIDITY_CONFIG_FILE_SIZE };
```
Comment (L-I-Am) [inline stack/apps/push7_button/filesystem/humidity_file.c]: best to allocate more

File: stack/apps/push7_button/filesystem/humidity_file.c [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+    ret = d7ap_fs_init_file(HUMIDITY_FILE_ID, &volatile_file_header, humidity_file.bytes);
+    if (ret != SUCCESS) {
+        log_print_error_string("Error initializing humidity effect file: %d", ret);
+    }
+
+    HDC1080DM_init(platf_get_i2c_handle());
+
+    d7ap_fs_register_file_modified_callback(HUMIDITY_CONFIG_FILE_ID, &file_modified_callback);
+    d7ap_fs_register_file_modified_callback(HUMIDITY_FILE_ID, &file_modified_callback);
+    d7ap_fs_read_file(HUMIDITY_CONFIG_FILE_ID, 0, humidity_config_file_cached.bytes, &length, ROOT_AUTH);
```
Comment (L-I-Am) [inline stack/apps/push7_button/filesystem/humidity_file.c]: don't we need to set a timer on the interval if enabled?

File: stack/apps/push7_button/filesystem/light_file.c [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+          .integration_time = ALS_INTEGRATION_100ms,
+          .persistence_protect_number = ALS_PERSISTENCE_1,
+          .gain = ALS_GAIN_x1,
+          .enabled = true };
+
+static light_config_file_t light_config_file_cached = (light_config_file_t) { .interval = DEFAULT_LIGHT_INTERVAL_SEC,
+    .integration_time = ALS_INTEGRATION_100ms,
+    .persistence_protect_number = ALS_PERSISTENCE_1,
+    .gain = ALS_GAIN_x1,
+    .enabled = true };
```
Comment (L-I-Am) [inline stack/apps/push7_button/filesystem/light_file.c:58]: can't you just initialise the cached variable with the default? or is this not possible?

File: stack/apps/push7_button/filesystem/light_file.c [hunk 11]
```diff
@@ -0,0 +1,193 @@
...
+        = { .file_permissions = (file_permission_t) { .guest_read = true, .user_read = true },
+              .file_properties.storage_class = FS_STORAGE_VOLATILE,
+              .length = LIGHT_FILE_SIZE,
+              .allocated_length = LIGHT_FILE_SIZE };
+
+    d7ap_fs_file_header_t permanent_file_header = { .file_permissions
+        = (file_permission_t) { .guest_read = true, .guest_write = true, .user_read = true, .user_write = true },
+        .file_properties.storage_class = FS_STORAGE_PERMANENT,
+        .length = LIGHT_CONFIG_FILE_SIZE,
+        .allocated_length = LIGHT_CONFIG_FILE_SIZE };
```
Comment (L-I-Am) [inline stack/apps/push7_button/filesystem/light_file.c]: best to allocate more

File: stack/apps/push7_button/filesystem/humidity_file.c [hunk 12]
```diff
@@ -0,0 +1,182 @@
...
+}
+
+static void file_modified_callback(uint8_t file_id)
+{
+    if (file_id == HUMIDITY_CONFIG_FILE_ID) {
+        uint32_t size = HUMIDITY_CONFIG_FILE_SIZE;
+        d7ap_fs_read_file(HUMIDITY_CONFIG_FILE_ID, 0, humidity_config_file_cached.bytes, &size, ROOT_AUTH);
+        if (humidity_config_file_cached.enabled) {
+            timer_cancel_task(&execute_measurement);
+            timer_post_task_delay(&execute_measurement, humidity_config_file_cached.interval * TIMER_TICKS_PER_SEC);
```
Comment (L-I-Am) [inline stack/apps/push7_button/filesystem/humidity_file.c]: a cancel and a post right after is not necessary anymore. Just use only post and it will overwrite the previous one (this is more efficient as well)

File: stack/apps/push7_button/filesystem/pir_file.c [hunk 13]
```diff
@@ -0,0 +1,199 @@
...
+        = { .file_permissions = (file_permission_t) { .guest_read = true, .user_read = true },
+              .file_properties.storage_class = FS_STORAGE_VOLATILE,
+              .length = PIR_FILE_SIZE,
+              .allocated_length = PIR_FILE_SIZE };
+
+    d7ap_fs_file_header_t permanent_file_header = { .file_permissions
+        = (file_permission_t) { .guest_read = true, .guest_write = true, .user_read = true, .user_write = true },
+        .file_properties.storage_class = FS_STORAGE_PERMANENT,
+        .length = PIR_CONFIG_FILE_SIZE,
+        .allocated_length = PIR_CONFIG_FILE_SIZE };
```
Comment (L-I-Am) [inline stack/apps/push7_button/filesystem/pir_file.c]: best to allocate more

File: stack/apps/push7_button/filesystem/pir_file.c [hunk 14]
```diff
@@ -0,0 +1,199 @@
...
+        platf_set_PIR_power_state(pir_config_file_cached.enabled && pir_file_transmit_state);
+        PYD1598_set_state(pir_config_file_cached.enabled && pir_file_transmit_state);
+        if (pir_config_file_transmit_state)
+            queue_add_file(pir_config_file_cached.bytes, PIR_CONFIG_FILE_SIZE, PIR_CONFIG_FILE_ID);
+    } else if (file_id == PIR_FILE_ID) {
+        pir_file_t pir_file;
+        uint32_t size = PIR_FILE_SIZE;
+        d7ap_fs_read_file(PIR_FILE_ID, 0, pir_file.bytes, &size, ROOT_AUTH);
+        if ((pir_file.mask == true && pir_config_file_cached.transmit_mask_1)
+            || (pir_file.mask == false && pir_config_file_cached.transmit_mask_0))
```
Comment (L-I-Am) [inline stack/apps/push7_button/filesystem/pir_file.c:132]: pir_file_transmit_state?

File: stack/apps/push7_button/filesystem/push7_state_file.c [hunk 15]
```diff
@@ -0,0 +1,176 @@
...
+        = { .file_permissions = (file_permission_t) { .guest_read = true, .user_read = true },
+              .file_properties.storage_class = FS_STORAGE_VOLATILE,
+              .length = PUSH7_STATE_FILE_SIZE,
+              .allocated_length = PUSH7_STATE_FILE_SIZE };
+
+    d7ap_fs_file_header_t permanent_file_header = { .file_permissions
+        = (file_permission_t) { .guest_read = true, .guest_write = true, .user_read = true, .user_write = true },
+        .file_properties.storage_class = FS_STORAGE_PERMANENT,
+        .length = PUSH7_STATE_CONFIG_FILE_SIZE,
+        .allocated_length = PUSH7_STATE_CONFIG_FILE_SIZE };
```
Comment (L-I-Am) [inline stack/apps/push7_button/filesystem/push7_state_file.c]: best to allocate more

File: stack/apps/push7_button/filesystem/push7_state_file.c [hunk 16]
```diff
@@ -0,0 +1,176 @@
...
+    ret = d7ap_fs_init_file(PUSH7_STATE_FILE_ID, &volatile_file_header, push7_state_file.bytes);
+    if (ret != SUCCESS) {
+        log_print_error_string("Error initializing push7_state effect file: %d", ret);
+    }
+    adc_stuff_init();
+    d7ap_fs_register_file_modified_callback(PUSH7_STATE_CONFIG_FILE_ID, &file_modified_callback);
+    d7ap_fs_register_file_modified_callback(PUSH7_STATE_FILE_ID, &file_modified_callback);
+
+    d7ap_fs_read_file(PUSH7_STATE_CONFIG_FILE_ID, 0, push7_state_config_file_cached.bytes, &length, ROOT_AUTH);
+    sched_register_task(&execute_measurement);
```
Comment (L-I-Am) [inline stack/apps/push7_button/filesystem/push7_state_file.c]: what is triggering the first measurement?

Comment (L-I-Am) [reply stack/apps/push7_button/filesystem/push7_state_file.c]: is it when push7_state_file_set_measure_state got called?

File: stack/apps/push7_button/sensor_manager.c [hunk 17]
```diff
@@ -0,0 +1,128 @@
...
+#include "stm32_common_gpio.h"
+
+#include "button_file.h"
+#include "hall_effect_file.h"
+#include "humidity_file.h"
+#include "light_file.h"
+#include "pir_file.h"
+#include "push7_state_file.h"
+
+#ifdef FRAMEWORK_NETWORK_MANAGER_LOG
```
Comment (L-I-Am) [inline stack/apps/push7_button/sensor_manager.c]: copy paste error?

File: stack/framework/hal/chips/HDC1080DM/HDC1080DM.h [hunk 18]
```diff
@@ -0,0 +1,35 @@
...
+	uint8_t rawData;
+	struct {
+		uint8_t HumidityMeasurementResolution : 2;
+		uint8_t TemperatureMeasurementResolution : 1;
+		uint8_t BatteryStatus : 1;
+		uint8_t ModeOfAcquisition : 1;
+		uint8_t Heater : 1;
+		uint8_t ReservedAgain : 1;
+		uint8_t SoftwareReset : 1;
+	};
```
Comment (L-I-Am) [inline stack/framework/hal/chips/HDC1080DM/HDC1080DM.h]: is this working? I would think it needs to be packed to map correctly

**Review by L-I-Am — CHANGES_REQUESTED** [review]

File: stack/framework/hal/platforms/PUSH7/platf_main.c [hunk 19]
```diff
@@ -70,93 +112,94 @@ void __platform_init()
...
 #ifdef PLATFORM_SX127X_USE_RESET_PIN
-  hw_gpio_configure_pin_stm(SX127x_RESET_PIN, &initStruct);
+    hw_gpio_configure_pin_stm(SX127x_RESET_PIN, &initStruct);
 #endif
-  hw_gpio_configure_pin_stm(SX127x_VCC_TXCO, &initStruct);
-
+    hw_gpio_configure_pin_stm(SX127x_VCC_TXCO, &initStruct);
 }
-
+       #define ROMBASE 0x1FF00000
```
Comment (L-I-Am) [inline stack/framework/hal/platforms/PUSH7/platf_main.c]: remove

File: stack/apps/push7_button/filesystem/pir_file.c (hunk 14 above)
Comment (KwintenSchram) [reply stack/apps/push7_button/filesystem/pir_file.c:132]: Pir sensor is actually turned off when transmit state is false, so checking it shouldn't be needed.

File: stack/apps/push7_button/filesystem/push7_state_file.c (hunk 16 above)
Comment (KwintenSchram) [reply stack/apps/push7_button/filesystem/push7_state_file.c]: yes, set_measure_state needs to be called for every sensor

File: stack/framework/hal/chips/HDC1080DM/HDC1080DM.h (hunk 18 above)
Comment (KwintenSchram) [reply stack/framework/hal/chips/HDC1080DM/HDC1080DM.h]: Haven't actually used this, so probably wouldn't have worked

File: stack/apps/push7_button/filesystem/humidity_file.c (hunk 9 above)
Comment (KwintenSchram) [reply stack/apps/push7_button/filesystem/humidity_file.c]: humidity_file_transmit_state needs to be set to true first using humidity_file_set_measure_state

File: stack/apps/push7_button/filesystem/button_file.c (hunk 4 above)
Comment (KwintenSchram) [reply stack/apps/push7_button/filesystem/button_file.c:134]: button_file_set_enabled is determined by the config file and can be turned off permanently that way, measure state is determined by the state of the firmware and isn't permanent. Alternate naming suggestions are welcome!

File: stack/framework/hal/chips/PYD1598/PYD1598.c [hunk 20]
```diff
@@ -0,0 +1,169 @@
...
+    struct {
+        uint32_t Factory_Params : 5; // Must be set to 0x10
+        uint32_t Filter_Source : 2; //
+        uint32_t Operation_Mode : 2;
+        uint32_t Window_Time : 2; // Window time = [RegisterValue] * 2s + 2s
+        uint32_t Pulse_Counter : 2; // Amount of pulses = [RegisterValue] + 1
+        uint32_t Blind_Time : 4; //[RegisterValue] *0.5s + 0.5s
+        uint32_t Threshold : 8; //
+        uint32_t reserved1 : 7;
+    };
```
Comment (L-I-Am) [inline stack/framework/hal/chips/PYD1598/PYD1598.c]: this is also not packed but using bits, is this fully working?

File: stack/framework/hal/chips/VEML7700/VEML7700.h [hunk 21]
```diff
@@ -0,0 +1,73 @@
...
+    struct {
+        uint16_t ALS_SD : 1; // ALS shut down setting
+        uint16_t ALS_INT_EN : 1; // ALS interrupt enable setting
+        uint16_t Reserved3 : 2;
+        uint16_t ALS_PERS : 2; // ALS persistence protect number setting
+        uint16_t ALS_IT : 4; // ALS integration time setting
+        uint16_t reserved2 : 1;
+        uint16_t ALS_GAIN : 2; // Gain selection
+        uint16_t reserved1 : 3;
+    };
```
Comment (L-I-Am) [inline stack/framework/hal/chips/VEML7700/VEML7700.h]: same here

File: stack/framework/hal/chips/VEML7700/VEML7700.h [hunk 22]
```diff
@@ -0,0 +1,73 @@
...
+} VEML7700_CONFIG_REG;
+
+typedef union {
+    uint8_t rawData;
+    struct {
+        
+        uint8_t PSM_EN : 1;
+        uint8_t PSM : 2;
+        uint8_t reserved : 7;
+    };
```
Comment (L-I-Am) [inline stack/framework/hal/chips/VEML7700/VEML7700.h]: and here

**Review by L-I-Am — APPROVED** [review]

File: stack/apps/push7_button/filesystem/hall_effect_file.c (hunk 7 above)
Comment (L-I-Am) [reply stack/apps/push7_button/filesystem/hall_effect_file.c]: now it gets checked on line 117 already so you can remove this one

File: stack/apps/push7_button/filesystem/humidity_file.c [hunk 23]
```diff
@@ -132,29 +137,29 @@ static void execute_measurement()
...
     if (test_mode_state == enable)
         return;
     test_mode_state == enable;
-    timer_cancel_task(&execute_measurement);
+    timer_cancel_task(&humidity_file_execute_measurement);
     if (enable) {
         humidity_config_file_cached.interval = TESTMODE_HUMID_INTERVAL_SEC;
         humidity_config_file_cached.enabled = true;
-        timer_post_task_delay(&execute_measurement, humidity_config_file_cached.interval * TIMER_TICKS_PER_SEC);
+        timer_post_task_delay(&humidity_file_execute_measurement, humidity_config_file_cached.interval * TIMER_TICKS_PER_SEC);
```
Comment (L-I-Am) [inline stack/apps/push7_button/filesystem/humidity_file.c:157]: didn't we agree testmode would now send data of a sensor upon button press? or is it not yet included here?

File: stack/apps/push7_button/push7_button.c [hunk 24]
```diff
@@ -112,55 +120,83 @@ static void switch_state(APP_STATE_t new_state)
...
     if (current_buttons_state == NO_BUTTON_PRESSED) {
-        if(max_buttons_state == NO_BUTTON_PRESSED)
+        if (max_buttons_state == NO_BUTTON_PRESSED)
             return;
-        if (max_buttons_state != prev_max_buttons_state) {
-            DPRINT("showing the state of %d",max_buttons_state );
-            display_state(sensor_enabled_state_array[max_buttons_state]);
 
-        } else {
+        if (max_buttons_state == prev_max_buttons_state) {
```
Comment (L-I-Am) [inline stack/apps/push7_button/push7_button.c:141]: I don't get this max_buttons_state

File: stack/apps/push7_button/push7_button.c [hunk 25]
```diff
@@ -169,22 +205,23 @@ static void app_state_input_event_handler(input_type_t i, bool mask)
...
 {
     little_queue_init();
-    led_flash_white();
     sched_register_task(&state_counter_event);
     button_file_register_cb(&userbutton_callback);
     booted_button_state = button_get_booted_state();
     sensor_manager_init();
 
-    if (booted_button_state == NO_BUTTON_PRESSED)
+    if (booted_button_state == NO_BUTTON_PRESSED) {
```
Comment (L-I-Am) [inline stack/apps/push7_button/push7_button.c]: change it to a switch case and set them in the right order?

File: stack/framework/hal/platforms/PUSH7/inc/led.h [hunk 26]
```diff
@@ -23,10 +23,10 @@
...
 #include "timer.h"
 
-#define FLASH_DURATION TIMER_TICKS_PER_SEC * 0.1
+#define FLASH_ON_DURATION TIMER_TICKS_PER_SEC * 0.1
+#define FLASH_OFF_DURATION TIMER_TICKS_PER_SEC * 0.2
 
 bool led_init(void);
-
-void led_flash_white(void);
+void start_led_flash(uint8_t flash_times);
```
Comment (L-I-Am) [inline stack/framework/hal/platforms/PUSH7/inc/led.h]: refactor to led_flash? don't know what "start_" adds to it

File: stack/framework/hal/platforms/PUSH7/platf_button.c [hunk 27]
```diff
@@ -62,11 +63,26 @@ __LINK_C void __ubutton_init()
...
+            booted_button_state += button_state << i;
+            buttons[i].last_known_state = button_state;
+        }
+    
     sched_register_task(&button_task);
+    sched_register_task(&validate_state);
+    sched_post_task(&validate_state);
+}
+
+static void validate_state()
```
Comment (L-I-Am) [inline stack/framework/hal/platforms/PUSH7/platf_button.c:78]: why is this here?

**Review by L-I-Am — CHANGES_REQUESTED** [review]

File: stack/framework/hal/platforms/PUSH7/platf_button.c (hunk 27 above)
Comment (KwintenSchram) [reply stack/framework/hal/platforms/PUSH7/platf_button.c:78]: When a button is pressed and released during reset, the button will be read out as high right after boot. When reading it out again afterwards, it will return the actual state of the GPIO.

File: stack/apps/push7_button/push7_button.c (hunk 24 above)
Comment (KwintenSchram) [reply stack/apps/push7_button/push7_button.c:141]: It's there to determine if multiple buttons were pressed at the same time.

File: stack/apps/push7_button/filesystem/humidity_file.c (hunk 23 above)
Comment (KwintenSchram) [reply stack/apps/push7_button/filesystem/humidity_file.c:157]: It is, see test_state_input_event_handler

**Review by L-I-Am — APPROVED** [review]
